### PR TITLE
Save recent covers to entityqueue and other improvements

### PR DESCRIPTION
**Pull request by nlisgo**
@jamiehollern

### CODE REVIEW

File: src/modules/jcms_migrate/src/EventSubscriber/JCMSMIgrateEventSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  /**
+   * Code to run after a row has been saved.
+   */
+  public function onPostRowSave(MigratePostRowSaveEvent $event) {
+    if ($event->getMigration()->id() == 'jcms_covers_db') {
+      if ($event->getRow()->getDestinationProperty('counter') <= 3) {
+        $subqueue = EntitySubqueue::load('covers');
+        $items = $subqueue->get('items')->getValue();
```
Comment (jamiehollern) [inline src/modules/jcms_migrate/src/EventSubscriber/JCMSMIgrateEventSubscriber.php]: For these sorts of things, I tend to be more defensive. Unless you're sure `getValue()` won't run on `null` I'd maybe separate this. It's not a huge issue though, if you're confident in it.

File: src/modules/jcms_rest/src/Plugin/rest/resource/PersonItemRestResource.php [hunk 2]
```diff
@@ -71,23 +71,39 @@ public function get($id) {
...
+          $expertises = [
+            'id' => [],
+            'name' => [],
+          ];
           foreach ($research_details_field->get('field_research_expertises') as $expertise) {
-            $research['expertises'][] = [
-              'id' => $expertise->get('entity')->getValue()->get('field_subject_id')->getString(),
-              'name' => $expertise->get('entity')->getValue()->toLink()->getText(),
-            ];
+            $expertise_id = $expertise->get('entity')->getValue()->get('field_subject_id')->getString();
```
Comment (jamiehollern) [inline src/modules/jcms_rest/src/Plugin/rest/resource/PersonItemRestResource.php]: Same as the above comment, this type of method chaining can be dangerous unless you're confident in it.

File: src/modules/jcms_migrate/src/EventSubscriber/JCMSMIgrateEventSubscriber.php (hunk 1 above)
Comment (nlisgo) [inline src/modules/jcms_migrate/src/EventSubscriber/JCMSMIgrateEventSubscriber.php]: Will make this adjustment. This one will certainly cause a problem.

File: src/modules/jcms_rest/src/Plugin/rest/resource/PersonItemRestResource.php (hunk 2 above)
Comment (nlisgo) [inline src/modules/jcms_rest/src/Plugin/rest/resource/PersonItemRestResource.php]: Will adjust

Comment (nlisgo) [inline src/modules/jcms_rest/src/Plugin/rest/resource/PersonItemRestResource.php]: Actually might refactor this later. This doesn't make the code any worse than it was and if we are going to introduce these checks we should probably be consistent. We could do this as a clean up step covered by a separate jira ticket.

**Pull request merged**